### PR TITLE
[Feature] add separated authorized user in audit log.

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

"SQL statements after `execute as xxx` statement, audit log still shows old user id as the user executing SQL statement. So that there is no easy way to ensure StarRocks uses a new user id to execute the SQL statement. I hope the user in the audit log can change to a new user id. Please help. Thanks."

Add a new field called `AuthroizedUser` to record the user used to authorize. The original `User` field merely represents the login name of the login user.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/plugin/AuditEvent.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|[]|

**Comment by padmejin** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Review by waittttting — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/plugin/AuditEvent.java [hunk 1]
```diff
@@ -55,8 +55,13 @@ public enum EventType {
     public long timestamp = -1;
     @AuditField(value = "Client")
     public String clientIp = "";
+    // The original login user
     @AuditField(value = "User")
     public String user = "";
+    // The user used to authorize
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/plugin/AuditEvent.java:61]: If userA->userB->userC, the impersonate list is a long list. If the userC executes the query, the print userB or userA?

**Review by murphyatwork — APPROVED** [review]

**Comment by padmejin** [comment]
@Mergifyio backport branch-2.4 branch-2.3
¡Muchas gracias!

**Pull request merged**